shremote.py

```diff
@@ -513,7 +522,17 @@ def safe_eval(st, label=''):
         log_error("Error evaluating expression '{}' due to [{}]".format(label, e))
         raise
 
+def try_eval(raw, label):
+    try:
+        evaled = safe_eval(raw, label)
+        if str(evaled) != str(raw): # Means that 'eval' did something
+            log_warn("Specifying evaluatable begin without $(...) is deprecated: {}".format(raw))
+        return float(evaled)
+    except TypeError: # raw shouldn't have to be eval'd again
+        return float(raw)
```

Comment:
Does this mean `$(...)` evals cannot return strings? It was only in the case of `begin` that I cast to a number. Other places, such as the strength of redos, I was using `$()` to expand strings